### PR TITLE
Update Krop + Various Build Improvements

### PR DESCRIPTION
**Pull request by noelwelsh**
This PR updates to Krop 0.6.0, and adds various improvements to the build:

- add and configure scalafix
- sbt-updates checks for plugin updates
- Add a "build" task that does everything we might want to do (compile, test, format, etc.)

### CODE REVIEW

File: backend/src/main/scala/snorri/Snorri.scala [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
 object Snorri {
-    val route =
-        Route(
-          Request.get(Path.root / "echo" / Param.string),
-          Response.ok[String]
-        ).handle(param => s"Your message was ${param}")
+  val route =
+    Route(
+      Request.get(Path.root / "echo" / Param.string),
+      Response.ok(Entity.text)
```
Comment (kannupriyakalra) [inline backend/src/main/scala/snorri/Snorri.scala]: @DmytroMitin this is a test comment to show how to review pull requests.

File: backend/src/main/scala/snorri/Snorri.scala [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
-import krop.all.{*, given}
+import krop.all.*
 
 object Snorri {
-    val route =
-        Route(
-          Request.get(Path.root / "echo" / Param.string),
-          Response.ok[String]
-        ).handle(param => s"Your message was ${param}")
+  val route =
```
Comment (kannupriyakalra) [inline backend/src/main/scala/snorri/Snorri.scala:34]: @zainab-ali another sample comment.

File: backend/src/main/scala/snorri/Snorri.scala (hunk 1 above)
Comment (DmytroMitin) [reply backend/src/main/scala/snorri/Snorri.scala]: @kannupriyakalra acknowledged, resolving

**Review by kannupriyakalra — APPROVED** [review]
LGTM

**Comment by kannupriyakalra** [comment]
Was getting this error and fixed this using below commands:

```
[warn] scalafmt: /home/runner/work/SnorriScalason/SnorriScalason/build.sbt isn't formatted properly!
[error] scalafmt: 1 files must be formatted (/home/runner/work/SnorriScalason/SnorriScalason)
[error] (Compile / scalafmtSbtCheck) scalafmt: 1 files must be formatted (/home/runner/work/SnorriScalason/SnorriScalason)
[error] Total time: 0 s, completed Nov 23, 2023 4:27:52 PM
```
sbt:root> scalafmtSbtCheck
sbt:root> scalafmtSbt

**Pull request closed without merging**